standalone/src/service.rs

```diff
@@ -184,131 +146,72 @@ pub fn new_full(mut config: Configuration) -> Result<(TaskManager, RpcHandlers),
         sc_service::build_offchain_workers(&config, task_manager.spawn_handle(), client.clone(), network.clone());
     }
 
-    let role = config.role.clone();
-    let force_authoring = config.force_authoring;
-    let backoff_authoring_blocks: Option<()> = None;
-    let name = config.network.node_name.clone();
-    let enable_grandpa = !config.disable_grandpa;
-    let prometheus_registry = config.prometheus_registry().cloned();
-
-    let rpc_extensions_builder = {
-        let client = client.clone();
-        let pool = transaction_pool.clone();
-
-        Box::new(move |deny_unsafe, _| {
-            let deps = interbtc_rpc::FullDeps {
-                client: client.clone(),
-                pool: pool.clone(),
-                deny_unsafe,
-            };
-
-            Ok(interbtc_rpc::create_full(deps))
-        })
-    };
+    // Proposer object for block authorship.
+    let env = sc_basic_authorship::ProposerFactory::new(
+        task_manager.spawn_handle(),
+        client.clone(),
+        transaction_pool.clone(),
+        config.prometheus_registry(),
+        None,
+    );
+
+    // Channel for the rpc handler to communicate with the authorship task.
+    let (mut command_sink, commands_stream) = mpsc::channel(10);
+
+    let rpc_sink = command_sink.clone();
+
+    let fut = transaction_pool.import_notification_stream().map(
```

Comment:
Maybe I'm missing something, but it doesn't look like this map is doing anything - `map` is lazy and doesn't do any work unless you do something with the result